app/src/layouts/map/worker/index.ts

```diff
@@ -0,0 +1,118 @@
+import { expose } from 'comlink';
+import { Buffer } from 'buffer';
+import { coordEach } from '@turf/meta';
+import wkx from 'wkx';
+
+type geometryOptions = {
+	geometryFormat?: GeometryFormat;
+	geometryField?: string;
+	longitudeField?: string;
+	latitudeField?: string;
+	geometrySRID?: string;
+};
+
+type GeometryFormat = 'geojson' | 'csv' | 'wkt' | 'wkb' | 'twkb' | 'lnglat';
+type GeometryParser = (entry: any) => wkx.Geometry | undefined;
+
+function lnglatParser(options: geometryOptions): GeometryParser {
+	return function (entry) {
+		const [lng, lat] = [entry[options.longitudeField!], entry[options.latitudeField!]];
+		return lng && lat && new wkx.Point(lng, lat);
+	};
+}
+function geojsonParser(options: geometryOptions): GeometryParser {
+	return function (entry) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parseGeoJSON(geom);
+	};
+}
+function twkbParser(options: geometryOptions): GeometryParser {
+	return function (entry) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parseTwkb(Buffer.from(geom, 'hex'));
+	};
+}
+function wktParser(options: geometryOptions): GeometryParser {
+	return function (entry) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parse(geom);
+	};
+}
+function wkbParser(options: geometryOptions): GeometryParser {
+	return function (entry) {
+		const geom = entry[options.geometryField!];
+		return geom && wkx.Geometry.parse(Buffer.from(geom, 'hex'));
+	};
+}
+function csvParser(options: geometryOptions): GeometryParser {
+	return function (entry) {
+		const geom = entry[options.geometryField!];
+		return geom && new wkx.Point(...geom.split(','));
+	};
+}
+
+function getGeometryParser(options: geometryOptions): GeometryParser {
+	switch (options.geometryFormat) {
+		case 'geojson':
+			return geojsonParser(options);
+		case 'lnglat':
+			return lnglatParser(options);
+		case 'twkb':
+			return twkbParser(options);
+		case 'wkb':
+			return wkbParser(options);
+		case 'wkt':
+			return wktParser(options);
+		case 'csv':
+			return csvParser(options);
+		default:
+			throw new Error('unimplemented format');
+	}
+}
+
+function expand(bbox: GeoJSON.BBox, coord: [number, number]) {
+	if (bbox[0] > coord[0]) bbox[0] = coord[0];
+	if (bbox[1] > coord[1]) bbox[1] = coord[1];
+	if (bbox[2] < coord[0]) bbox[2] = coord[0];
+	if (bbox[3] < coord[1]) bbox[3] = coord[1];
+}
+
+type AllGeoJSON = GeoJSON.FeatureCollection & GeoJSON.Feature & GeoJSON.GeometryCollection & GeoJSON.Geometry;
+
+function toGeoJSON(
+	entries: any[],
+	options: geometryOptions,
+	onProgress: (p: number) => void
+): GeoJSON.FeatureCollection {
+	const parser = getGeometryParser(options);
+	const project = (coord: any): any => coord; // TODO
+	const geojson: GeoJSON.FeatureCollection = {
```

Comment:
@Oreilles What where you planning to do here?

Reply:
Handling the te projection of coordinates from any CRS to WGS84, which is the standard for GeoJSON, using the library proj4js.

Reply:
And why? 😅

Reply:
Because not everyone use WGS84 😉 There are many countries in which public geospatial data is distributed in the CRS that makes the most sense for said country. In France for example, most datasets provided by the IGN are in EPSG:2154 because it allows for the least amount of distortion over France when viewed on a 2D map.

Reply:
Is this a feature that you think is nessesary before we're going to merge this?
Also, is there a way to find out what projection was used or does this has to be configured by the user?

Reply:
There isn't a way to guess from the data only (there are thousands of different CRS), so it should indeed be configured by the user (defaulting to WGS84 if unspecified). I think it's a very important feature, as geospatial data is basically unusable if in a different CRS (it will be shown on the map at wrong locations, or nowhere at all if the projections are too different) - plus it's not that much work to implement as there already exist a library doing that (proj4). 

But it's true that most non-specialists only deal with WGS84 data, so ultimately I don't think it is absolutely necessary for a v1.